src/App.tsx

```diff
@@ -176,67 +226,83 @@ function App() {
     engine.interruptGenerate();
   }
 
+  const [sidebarOpen, setSidebarOpen] = useState(false);
```

Comment:
Let's move this to useChatStore so we don't have to do prop drilling. I'm using a single zustand global store in useChatStore.